packages/compass-import-export/src/modules/export.ts

```diff
@@ -281,12 +284,22 @@ export const selectFieldsToExport = (): ExportThunkAction<
     });
 
     const {
-      export: { query, namespace },
+      export: { query, namespace, connectionId },
     } = getState();
 
     let gatherFieldsResult: Awaited<ReturnType<typeof gatherFieldsFromQuery>>;
 
     try {
+      if (!connectionId) {
+        throw new Error('ConnectionId not provided');
+      }
+
+      const dataService =
+        connectionsManager.getDataServiceForConnection(connectionId);
+      if (!dataService) {
+        throw new Error(`DataService for connection ${connectionId} not found`);
+      }
```

Comment:
Is there any case where we don't want this to throw? Is it something we can include as part of the get implementation?